### PR TITLE
Added button to invite to channel

### PR DESCRIPTION
**Pull request by maxer137**
Related to #38

I do not know if there's a "manage channel" view that we can navigate to. 
For now, I left the "manage channel" button commented out.

### CODE REVIEW

**Comment by Rexogamer** [comment]
It seems `open_channel_settings` is the action in question (although instead of opening a popup it's a separate page (`channel/<id>/settings`)).

File: src/components/navigation/items/ButtonItem.tsx [hunk 1]
```diff
@@ -1,8 +1,9 @@
-import { X, Crown } from "@styled-icons/boxicons-regular";
+import { X, Crown, UserPlus } from "@styled-icons/boxicons-regular";
 import { observer } from "mobx-react-lite";
 import { Presence } from "revolt-api/types/Users";
 import { Channel } from "revolt.js/dist/maps/Channels";
 import { User } from "revolt.js/dist/maps/Users";
+import { ChannelPermission } from "revolt.js/dist/api/permissions";
```
Comment (Rexogamer) [inline src/components/navigation/items/ButtonItem.tsx:6]: Imports:
```suggestion
import { X, Crown, UserPlus, Cog } from "@styled-icons/boxicons-regular";
import { observer } from "mobx-react-lite";
import { Link } from "react-router-dom";
import { Presence } from "revolt-api/types/Users";
import { ChannelPermission } from "revolt.js/dist/api/permissions";
import { Channel } from "revolt.js/dist/maps/Channels";
import { User } from "revolt.js/dist/maps/Users";
```

File: src/components/navigation/items/ButtonItem.tsx [hunk 2]
```diff
@@ -207,6 +208,33 @@ export const ChannelButton = observer((props: ChannelProps) => {
...
+                {/*        className={styles.icon}*/}
+                {/*        onClick={() =>*/}
+                {/*            openScreen({*/}
+                {/*                id: "channel_info",*/}
+                {/*                channel: channel,*/}
+                {/*            })*/}
+                {/*        }>*/}
+                {/*        <Cog size={24} />*/}
+                {/*    </IconButton>*/}
+                {/*)}*/}
```
Comment (Rexogamer) [inline src/components/navigation/items/ButtonItem.tsx:237]: Button code:
```suggestion
                {!isTouchscreenDevice &&
                    channel.channel_type === "TextChannel" &&
                    channel.permission & ChannelPermission.ManageChannel && (
                        <div className="actions">
                            <Link to={`/channel/${channel._id}/settings`}>
                                <IconButton>
                                    <Cog size={24} />
                                </IconButton>
                            </Link>
                        </div>
                    )
                }
```

**Review by Rexogamer — CHANGES_REQUESTED** [review]
Here's the manage channel code. One thing I'd like to note is that it seems both of these buttons appear even if you lack the relevant perms, so some further debugging will be needed there.

**Comment by maxer137** [comment]
> Here's the manage channel code. One thing I'd like to note is that it seems both of these buttons appear even if you lack the relevant perms, so some further debugging will be needed there.

I haven't been able to reproduce this. For me, the buttons only show up when you have the right permissions.

**Comment by ryanalexander** [comment]
@maxer137 are you still working on this? If not i will happily take over the feature

**Comment by Rexogamer** [comment]
Just checking in - do you still want to work on this?

**Pull request closed without merging**